Only sanitize `content` attribute when present in attachments

p8 opened the pull request:
When the `content` attribute is set for Action Text attachments, Trix shows this content.

1ac6d40d36a07b48a67bc7f8627fd1f92bffcb14 introduced sanitizing the `content` attribute of ActionText::Attachable::ContentAttachment. 
However, it would also set the `content` attribute when it isn't present.
So, instead of showing the image preview, Trix would use the empty `content` attribute resulting in missing previews for images.

An issue was reported in the Basecamp repo: https://github.com/basecamp/trix/issues/1158

### Additional information

The test that was added in 1ac6d40d36a07b48a67bc7f8627fd1f92bffcb14 did
not fail when commenting out the original sanitation fix. I've added a test to fix this as well.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

p8 committed:
Only sanitize `content` attribute when present in attachments

When the `content` attribute is set for Action Text attachments, Trix
shows this content.

1ac6d40d36a07b48a67bc7f8627fd1f92bffcb14 introduced sanitizing the
`content` attribute of ActionText::Attachable::ContentAttachment.
However, it would also set the `content` attribute when it isn't
present. Instead of showing the image preview, Trix would use the
empty `content` attribute resulting in missing previews for images.

rafaelfranca commented:
Can't we write an unit test just to make sure the `content` node is present in the generated HTML?

rafaelfranca commented:
How does this fix the issue. I just remove the `node["content"] = sanitize_content_attachment(node["content"])` call and all tests pass, so I'm not even sure if that call is necessary and what it does.

p8 commented:
@rafaelfranca I've added a test to reproduce the issue.
The test that was added in https://github.com/rails/rails/commit/1ac6d40d36a07b48a67bc7f8627fd1f92bffcb14 did not fail when commenting out the original sanitation fix. 
I've added a test to fix this as well.
